### DISCUSSION TITLE
How to include JS file from node_modules in project?

### DISCUSSION THREAD

**Discussion by swati61** [vercel/next.js#40012] [Help]
I have following code to load a scripts.min.js file from https://www.npmjs.com/package/@comp/web-sdk which require initialization on application load. but I am getting error ReferenceError: self is not defined

`function MyApp({ Component, pageProps }) {
  return (
    <>
    <Head>
        <title>First Post</title>
      </Head>
      <Script
        src="/node_modules/@comp/web-sdk/js/scripts.min.js"
        onLoad={() =>{
          console.log("Here")
          window.WS.formElements.initSelect()
        }
        }
      />
        <Component {...pageProps} />
    </>
);
}

export default MyApp`

**Comment by ghost** [comment]
You need to move this file into your `public` directory at build time so that it falls inline with the other patterns already established by this framework. Here's a reference for you: https://stackoverflow.com/questions/71887731/next-js-best-way-to-serve-static-js-from-a-node-modules-dist-folder

**Comment by pjuangph** [comment]
I overcame this by using Flowbite-react 
The setup is easy and there's no need to include any javascript in the `<body> <script src="cdn Flowbite" > </script> </body>`

See: 
[Flowbite-react dropdown example](https://www.flowbite-react.com/docs/components/dropdown)

There is a mistake in their docs though. You want to use `DropdownItem` instead of `Dropdown.item`
